ably/realtime_channel.go

```diff
@@ -548,6 +547,17 @@ func (c *RealtimeChannel) SubscribeAll(ctx context.Context, handle func(*Message
 	return unsubscribe, nil
 }
 
+// SubscribeAll registers an event listener for error messages on this channel.
+//
+// See package-level documentation => [ably] Event Emitters for details about messages dispatch.
+func (c *RealtimeChannel) OnError(handle func(*ErrorInfo)) (func(), error) {
+	// unsubscribe deregisters all listeners to error messages on this channel.
+	unsubscribe := c.errorEmitter.OnAll(func(err emitterData) {
+		handle((*ErrorInfo)(err.(*errorMessage)))
+	})
+	return unsubscribe, nil
```

Comment:
Given this function never returns an error, perhaps drop the `err` return value?